xapian-core/include/xapian/diversify.h

```diff
@@ -90,6 +92,7 @@ class XAPIAN_VISIBILITY_DEFAULT Diversify {
      *	  			[1e-6,1]
      */
     explicit Diversify(unsigned int k_,
+		       double r_,
```

Comment:
The lack of a default value for `r_` here is causing the tests to fail (in the api_generated tests, because those expect that `Xapian::Diversify foo(10);` works).

If `r_` is a number of document, an integer type would seem more appropriate (e.g. `Xapian::doccount` - perhaps `k_` should be that too actually).

Reply:
Oops, forgot to modify the api test for ```Xapian::Diversify```. And also thanks for pointing out ```Xapian::doccount```, I didn't realise that was there. Pushed two fixup commits to reflect these changes.